src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+#include "ac_uki.h"
+#include "ac_npc_anim_def.h"
+
+static void aNSM_actor_ct(ACTOR *shop_master, GAME* game);
+static void aNSM_actor_dt(ACTOR* actorx, GAME* game);
+static void aNSM_actor_move(ACTOR* actorx, GAME* game);
+static void aNSM_actor_draw(ACTOR* actorx, GAME* game);
+static void aNSM_actor_init(ACTOR* actorx, GAME* game);
+static void aNSM_actor_save(ACTOR* actorx, GAME* game);
+
+static void aNSM_talk_request(ACTOR* actorx, GAME* game);
+static int aNSM_talk_init(ACTOR* actorx, GAME* game);
+static int aNSM_talk_end_chk(ACTOR* actorx, GAME* game);
+
+static void aNSC_BGcheck(ACTOR *actorx);
+static void aNSC_set_zone_data(NPC_SHOP_MASTER_ACTOR *shop_master, ACTOR *actor);
+static void aNSC_set_player_angl(NPC_SHOP_MASTER_ACTOR *shop_master);
+static void aNSC_talk_demo_proc(ACTOR *actorx);
+static void aNSC_sell_camera(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game);
+
+static void aNSM_actor_ct(ACTOR *actorx, GAME* game) {
+    static aNPC_ct_data_c ct_data = {
+        &aNSM_actor_move,
+        &aNSM_actor_draw,
+        0,
+        &aNSM_talk_request,
+        &aNSM_talk_init,
+        &aNSM_talk_end_chk,
+        TRUE 
+    };
+
+    NPC_SHOP_MASTER_ACTOR *shop_master = (NPC_SHOP_MASTER_ACTOR *)actorx;
+    int action = 0;
+
+    CLIP(npc_clip)->ct_proc(actorx, game, &ct_data);
+
+    shop_master->npc_class.draw.main_animation.keyframe.morph_counter = 0.0f;
+    shop_master->sell_item = EMPTY_NO;
+    shop_master->npc_class.condition_info.hide_flg = FALSE;
+    shop_master->_9C0 = -1;
+    actorx->shape_info.draw_shadow = TRUE;
+
+    if (Common_Get(door_data).door_actor_name == RSV_NO) {
+        shop_master->npc_class.talk_info.melody_inst = 0;
+        action = 0x3d; 
+    } else {
+        action = 0;
+    }
+
+    aNSC_setupAction(shop_master, (GAME_PLAY *)game, action);
+}
+
+static void aNSM_actor_save(ACTOR* actorx, GAME* game) {
+    (*Common_Get(clip).npc_clip->save_proc)(actorx, game);
+}
+
+static void aNSM_actor_dt(ACTOR* actorx, GAME* game) {
+    (*Common_Get(clip).npc_clip->dt_proc)(actorx, game);
+}
+
+static void aNSM_actor_init(ACTOR* actorx, GAME* game) {
+    (*Common_Get(clip).npc_clip->init_proc)(actorx, game);
+}
+
+static void aNSM_actor_draw(ACTOR* actorx, GAME* game) {
+    (*Common_Get(clip).npc_clip->draw_proc)(actorx, game);
+}
+
+static int aNSM_get_zone(xyz_t *wpos) {
+    if (wpos->x < (f32)120.0) {
+        if (wpos->z < (f32)180.0) {
+            return 1;
+        } else {
+            return 2;
+        }
+    }
+    if (wpos->z < (f32)160.0) {
+        return 0;
+    } else {
+        return 3;
+    }
+}
+
+static u8 aNSM_get_next_zone(u8 p1, u8 p2) {
+    static u8 next_zone[16] = {0, 1, 1, 1, 0, 1, 2, 2, 1, 1, 2, 3, 2, 2, 2, 3};
+    int t = p2 * 4;
+    return next_zone[t + p1];
+}
+
+static void aNSM_search_player(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *play) {
+    ACTOR *actorx = (ACTOR *)shop_master;
+    PLAYER_ACTOR *player = get_player_actor_withoutCheck(play); 
```

Comment:
We shouldn't be using get_player_actor_withoutCheck directly. Please use the `GET_PLAYER_ACTOR_*` macros.

Reply:
The 0x42 in the call needs to be made into an enum. These are the action states and you can find suitable names in the function pointer table in aNSC_setupAction.